pydicom/filebase.py

```diff
@@ -30,19 +30,13 @@ def __init__(self, *args, **kwargs):
     def read_le_tag(self):
         """Read and return two unsigned shorts (little endian)
            from the file."""
-        bytes_read = self.read(4)
-        if len(bytes_read) < 4:
-            # needed for reading "next" tag when at end of file
-            raise EOFError
+        bytes_read = self.read(4, need_exact_length=True)
         return unpack(b"<HH", bytes_read)
 
     def read_be_tag(self):
```

Comment:
> Given that there are more API breaking changes in 1.0, all pydicom users know about it 

I would say that this the purpose to go from 0.x.x to 1.x.x, there no more back-compatibility imposed. You can introduce the deprecation within the current major version such that what ever change in 1.0.x have to be compatible with 1.0.x-2